quotefault/templates/bootstrap/storage.html

```diff
@@ -18,10 +18,53 @@
             {% endif %}
         {% endwith %}
         {{ display( quotes ) }}
-        <button href="#moreQuotes" id="get_more" data-toggle="collapse" class="btn btn-default center-block">But wait, there's more!</button>
-        <br>
-        <div id="moreQuotes" class="collapse" aria-expanded="false">
-        </div>
+        <nav aria-label="page selector">
+            <ul class="pagination pagination-lg justify-content-center">
+                {% if page == 1 %}
+                <li class="page-item disabled"> 
+                    <a class="page-link" href="/storage/{{ page - 1 }}" tabindex="-1">Previous</a>
+                {% else %}
+                <li class="page-item">
+                    <a class="page-link" href="/storage/{{ page - 1 }}">Previous</a>
+                {% endif %}
+                </li>
+                {% if rows <= 12 %}
+                    {% for num in range(1, rows + 1) %}
+                        <li class="page-item">
+                            <a class="page-link" href="/storage/{{ num }}">{{ num }}</a>
+                        </li>
+                    {% endfor %}
+                {% elif page < 6 %}
+                    {% for num in range(1, 12) %}
+                        <li class="page-item">
+                            <a class="page-link" href="/storage/{{ num }}">{{ num }}</a>
+                        </li>
+                    {% endfor %}
+                {% elif rows - page <= 5 %}
+                    {% for num in range(rows - 10, rows + 1) %}
+                    <li class="page-item">
+                        <a class="page-link" href="/storage/{{ num }}">{{ num }}</a>
+                    </li>
+                    {% endfor %}
+                {% else %}
+                    {% for num in range(page - 5, page + 6) %}
+                        <li class="page-item">
+                            <a class="page-link" href="/storage/{{ num }}">{{ num }}</a>
+                        </li>
+                    {% endfor %}
+                {% endif %}
```

Comment:
Why is this logic so complicated? can this be `for num in range(max(1, page-6), min(page+6, rows) + 1)` or similar?